src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -0,0 +1,424 @@
+    public void iCanAssertThatRedemptionWasSuccessfulByAssertingTheStatusInResponse() {
+        try {
+            JsonNode rootNode = objectMapper.readTree(redeemVoucherResponseBody);
+
+            String status = rootNode.get("status").asText();
+            System.out.println("Status: " + status);
+            assertThat(status).isEqualTo("01");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+    public void assertUnsuccessfulRedemption(){
+        try {
+            JsonNode rootNode = objectMapper.readTree(redeemVoucherResponseBody);
+
+            String status = rootNode.get("status").asText();
+            System.out.println("Status: " + status);
+            assertThat(status).isEqualTo("00");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    @Before("@createAndActivateVoucher")
+    public void createAndActivateVoucher(){
+        System.out.println("\n\n\n\n I am running before all \n\n\n\n");
+        iCreateAnIdentityMapperDTOForRegisterBeneficiary();
+        mockServerStepDef.checkIfMockServerIsInjected();
+        mockServerStepDef.startMockServer();
+        iCallTheVoucherCreateAPIWithExpectedStatusOf(202, "/createVoucher");
+        mockServerStepDef.startStub("/createVoucher", PUT, 200);
+        mockServerStepDef.hitStubEndpoint(PUT, "/createVoucher", 200);
+        iShouldBeAbleToExtractResponseBodyFromCallback();
+        mockServerStepDef.stopMockServer();
+        iCanCreateAnVoucherRequestDTOForVoucherActivation();
+        mockServerStepDef.checkIfMockServerIsInjected();
+        mockServerStepDef.startMockServer();
+        iCallTheActivateVoucherAPIWithExpectedStatusOfAndStub(202, "/activateVoucher");
+        mockServerStepDef.startStub("/activateVoucher", PUT, 200);
+        mockServerStepDef.hitStubEndpoint(PUT, "/activateVoucher", 200);
+        iShouldBeAbleToVerifyThatTheMethodToEndpointReceivedARequestWithRequiredParameterInActivateVoucherCallbackBody("PUT", "/activateVoucher");
+        mockServerStepDef.stopMockServer();
+    }
+    @After("@redeemVoucherFailure")
+    public void redeemVoucherFailure(){
+        iCanCreateAnRedeemVoucherRequestDTOForVoucherRedemption();
+        iCallTheRedeemVoucherAPIWithExpectedStatusOf(200);
+        assertUnsuccessfulRedemption();
+    }
+    @After("@redeemVoucherSuccess")
+    public void redeemVoucherSuccess(){
+        iCanCreateAnRedeemVoucherRequestDTOForVoucherRedemption();
+        iCallTheRedeemVoucherAPIWithExpectedStatusOf(200);
+        iCanAssertThatRedemptionWasSuccessfulByAssertingTheStatusInResponse();
+    }
+
+    @Given("I can create an VoucherRequestDTO for voucher suspension")
+    public void iCanCreateAnVoucherRequestDTOForVoucherSuspension() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("{\n");
+        sb.append("    \"requestID\": \"849324499155\",\n");
+        sb.append("    \"batchID\": \"045155518258\",\n");
```

Comment:
arent these unique? this hardcoding will lead to failure when running multiple times. so use generated values.

Reply:
Removed hardcoding